src/integrations.py

```diff
@@ -0,0 +1,83 @@
+import hashlib
+import logging
+from dataclasses import dataclass
+from secrets import token_bytes
+from typing import Optional
+
+from charms.glauth_k8s.v0.ldap import LdapProviderBaseData, LdapProviderData
+from configs import DatabaseConfig
+from constants import DEFAULT_GID, DEFAULT_UID, GLAUTH_LDAP_PORT
+from database import Capability, Group, Operation, User
+from ops.charm import CharmBase
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class BindAccount:
+    cn: Optional[str] = None
+    ou: Optional[str] = None
+    password: Optional[str] = None
+
+
+def _create_bind_account(dsn: str, user_name: str, group_name: str) -> BindAccount:
+    with Operation(dsn) as op:
+        if not op.select(Group, Group.name == group_name):
+            group = Group(name=group_name, gid_number=DEFAULT_GID)
+            op.add(group)
+
+        if not (user := op.select(User, User.name == user_name)):
+            new_password = hashlib.sha256(token_bytes()).hexdigest()
```

Comment:
I chose to create a random SHA256-hashed password for the initial password of the bind account. The end user can later use the `glauth-utils` charmed operator to modify it if they would like to do it.